File: bookOfProblems/leetcode.com/c++/121. Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int min_price = INT32_MAX;
		int max_profit = INT32_MIN;
		
		for(auto& i : prices) {
			min_price = std::min(min_price, i);
			max_profit = std::max(max_profit, i - min_price);
		}
		
		return max_profit;
    }
};

File: bookOfProblems/leetcode.com/golang/121. Best Time to Buy and Sell Stock.go
func maxProfit(prices []int) int {
	maxProfit := 0
	currentProfit := 0
	windowStart := 0
	windowEnd := 0
	
	for windowEnd < len(prices) {
		if prices[windowStart] > prices[windowEnd] {
			windowStart = windowEnd
		} else {
			currentProfit = prices[windowEnd] - prices[windowStart]
			maxProfit = max(maxProfit, currentProfit)
		}
		windowEnd++
	}

	return maxProfit
}